### PR TITLE
Cleaning IbisITK directory

### PR DESCRIPTION
**Pull request by hgueziri**
Moved directories from IbisITK to their respective plugins
Deleted IbisITK directory

- itkSamplers, itkRegistration and itkRegistrationOpenCL are now located in IbisPlugins/GPU_RigidRegistration directory
- itkVolumeReconstructionOpenCL is now located in IbisPlugins/GPU_VolumeReconstruction directory

- Cleaned itkRegistration directory to remove unused files.
- Moved the following files from itkRegistration to itkSamplers for consistency as they are used for image sampling:
-- itkImageToVectorContainerFilter.h
-- itkImageToVectorContainerFilter.txx
-- itkVectorContainerSource.h
-- itkVectorContainerSource.txx
-- itkVectorDataContainer.h
-- itkVectorDataContainer.txx
- Cleaned itkRegistrationOpenCL directory to remove unused files.

Compiles in Linux with the GPU_VolumeReconstruction, the GPU_registration and the USManualCalibration plugins ON
Need further tests for compilation in Mac

### CODE REVIEW

**Comment by drouin-simon** [comment]
I just tested the build on Mac. No problem.

Only thing to fix: the IbisITK folder is empty but the folder is still there. Can you remove it before we merge the PR?

**Comment by hgueziri** [comment]
When compiling without Registration plugin, the compilation fails. I fixed the problem by moving CMake declaration of OpenCL from Ibis/CMakeLists to GPU_Registration/CMakeLists. I did the same for the GPU_VolumeReconstruction plugin.
I also duplicated the itkOpenCLUtil.h .cxx in itkVolumeReconstructionOpenCL so that each plugin can be built independently.

I don't have an empty IbisITK folder in my fork. How to fix it?

File: IbisPlugins/GPU_VolumeReconstruction/CMakeLists.txt [hunk 1]
```diff
@@ -19,3 +19,7 @@ ENDIF()
 # Create plugin
 DefinePlugin( "${PluginSrc}" "${PluginHdr}" "${PluginHdrMoc}" "${PluginUi}" )
 target_link_libraries( ${PluginName} itkVolumeReconstructionOpenCL )
+
+IF( OPENCL_FOUND )
+    set( itkLibsToUse ${itkLibsToUse} itkVolumeReconstructionOpenCL)
+ENDIF()
```
Comment (drouin-simon) [inline IbisPlugins/GPU_VolumeReconstruction/CMakeLists.txt]: Houssem, en principe, ces lignes ne devraient pas être nécessaires puisque tu as déjà:
```
target_link_libraries( ${PluginName} itkVolumeReconstructionOpenCL ) 
```
plus haut, donc, n'importe quel exécutable qui link avec la librairie du plugin va aussi linker avec itkVolumeReconstructionOpenCL.

File: IbisPlugins/GPU_RigidRegistration/CMakeLists.txt [hunk 2]
```diff
@@ -26,3 +26,8 @@ DefinePlugin( "${PluginSrc}" "${PluginHdr}" "${PluginHdrMoc}" "${PluginUi}" )
 #add library specific to this plugin
 target_link_libraries( ${PluginName} itkSamplers itkRegistration itkRegistrationOpenCL )
 
+set( itkLibsToUse ${itkLibsToUse} itkRegistration )
+set( itkLibsToUse ${itkLibsToUse} itkSamplers )
+IF( OPENCL_FOUND )
+    set( itkLibsToUse ${itkLibsToUse} itkRegistrationOpenCL)
+ENDIF()
```
Comment (drouin-simon) [inline IbisPlugins/GPU_RigidRegistration/CMakeLists.txt]: Même commentaire que pour IbisPlugins/GPU_VolumeReconstruction/CMakeLists.txt

**Pull request merged**